### PR TITLE
work for issue #379 - adding back rpm and deb tests

### PR DESCRIPTION
**Pull request by EricDavisX**
## What does this PR do?
2 things:
1) very small change to rename the @ tag to be more accurate per the tests being done and to sync 2 file names to the @ tag name for the whole file (I think this is our best practice pattern?)

2) most primarily, this is work for https://github.com/elastic/e2e-testing/issues/379 - adding in the deb and rpm tests against centos & debian images that were removed prior.

I used existing .go implementation code that remained and which I think is functional and updated so far, only feature files.

## Why is it important?
we need coverage over rpm and deb installs, even tho they are no longer the recommended usage - they are still supported and expected to be used.

In 379 you can see the discussion, we removed this accidentally / temporarily as we implemented new support for 'install' feature.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have run the Unit tests for the CLI, and they are passing locally~~
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally~~
~~- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)~~

## Author's Checklist

I actually am opening this as a draft to get some help from @mdelapenya or perhaps @blakerouse if we need

## How to test this PR locally
OP_LOG_LEVEL=DEBUG DEVELOPER_MODE=true godog -t enroll > test-debug.log

## Related issues

- Relates https://github.com/elastic/beats/pull/22543/files

## Follow-ups

So, Centos is working as submitted here,  But Debian is throwing an error.. and I am not 100% sure I'm tracing the go code we've implemented right to know what question to ask.  I'll put logs and more details in the comments and we can figure it out I hope.

### CODE REVIEW

**Comment by EricDavisX** [comment]
@mdelapenya when I run Centos, it works super.  But when I run Debian, it throws an error and I can't trace the logging or the go code itself to be sure.  I could use help, please.

The error cited in the logs looks to relate exactly to the error we were attempting to fix in the cited issue above.  perhaps it is just not fixed yet?  But also, perhaps, our code needs some modification.  I'd like to clarify what it is intending tho...  we use a Debian image, with systemd - but the scripting indicates it is using some sysv init usage so, I'm not sure what is going on.  

```
INFO[0158] Kibana setup initialised                      responseBody="{\"isReady\":true,\"missing_requirements\":[]}"
DEBU[0159] Fleet token created                           apiKeyId=e4Jq3XUBvCjQ32uaszD6 tokenId=34a21e03-8491-428b-a512-01668beaa9f3
Pulling debian-systemd (alehaa/debian-systemd:stretch)...
fleet_package-registry_1 is up-to-date
fleet_elasticsearch_1 is up-to-date
Creating fleet_debian-systemd_elastic-agent_1 ... 
Creating fleet_debian-systemd_elastic-agent_1 ... done
DEBU[0166] Docker compose executed.                      cmd="[up -d]" composeFilePaths="[/Users/edavis/.op/compose/profiles/fleet/docker-compose.yml /Users/edavis/.op/compose/services/debian-systemd/docker-compose.yml]" env="map[debian_systemdAgentBinarySrcPath:/var/folders/k7/plkh2nd96w54yd1ytt3dl8br0000gq/T/elastic-agent-8.0.0-SNAPSHOT-amd64.deb848641955 debian_systemdAgentBinaryTargetPath:/elastic-agent-8.0.0-SNAPSHOT-amd64.deb debian_systemdContainerName:fleet_debian-systemd_elastic-agent_1 debian_systemdTag:stretch kibanaConfigPath:/Users/edavis/e2e-testing/e2e/_suites/fleet/configurations/kibana.config.yml stackVersion:8.0.0-SNAPSHOT]" profile=fleet

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

debconf: delaying package configuration, since apt-utils is not installed
DEBU[0169] Docker compose executed.                      cmd="[exec -T debian-systemd apt install /elastic-agent-8.0.0-SNAPSHOT-amd64.deb -y]" composeFilePaths="[/Users/edavis/.op/compose/profiles/fleet/docker-compose.yml /Users/edavis/.op/compose/services/debian-systemd/docker-compose.yml]" env="map[debian_systemdAgentBinarySrcPath:/var/folders/k7/plkh2nd96w54yd1ytt3dl8br0000gq/T/elastic-agent-8.0.0-SNAPSHOT-amd64.deb848641955 debian_systemdAgentBinaryTargetPath:/elastic-agent-8.0.0-SNAPSHOT-amd64.deb debian_systemdContainerName:fleet_debian-systemd_elastic-agent_1 debian_systemdTag:stretch kibanaConfigPath:/Users/edavis/e2e-testing/e2e/_suites/fleet/configurations/kibana.config.yml stackVersion:8.0.0-SNAPSHOT]" profile=fleet
Synchronizing state of elastic-agent.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable elastic-agent
update-rc.d: error: elastic-agent Default-Start contains no runlevels, aborting.
ERRO[0170] Could not execute command in container        command="[systemctl enable elastic-agent]" error="Could not run compose file: [/Users/edavis/.op/compose/profiles/fleet/docker-compose.yml /Users/edavis/.op/compose/services/debian-systemd/docker-compose.yml] - Local Docker compose exited abnormally whilst running docker-compose: [exec -T debian-systemd systemctl enable elastic-agent]. exit status 1" service=debian-systemd
ERRO[0170] Could not enable the service                  command="[systemctl enable elastic-agent]" error="Could not run compose file: [/Users/edavis/.op/compose/profiles/fleet/docker-compose.yml /Users/edavis/.op/compose/services/debian-systemd/docker-compose.yml] - Local Docker compose exited abnormally whilst running docker-compose: [exec -T debian-systemd systemctl enable elastic-agent]. exit status 1" service=debian-systemd
WARN[0170] Could not create command in container         command="[cat /etc/elastic-agent/.elastic-agent.active.commit]" container= detach=false error="Error response from daemon: page not found" tty=false
ERRO[0170] Could not get agent hash in the container     containerName= error="Error response from daemon: page not found"

```

File: e2e/_suites/fleet/features/fleet_mode_agent.feature
Comment (EricDavisX) [inline e2e/_suites/fleet/features/fleet_mode_agent.feature:18]: FYI -  I wanted to keep this separate from the above 'install' step due to the underlying sub-command tested.  it would be a different failure if this broke but the other did not.

**Comment by EricDavisX** [comment]
also - I don't know how to re-trigger tests apart from making a nominal change to a file in the pr.  do we not support something like /package or /git-bot test this pr for me pretty please?

I have no idea why a pre-submit test is failing.  0_o  Instruct me, so I may pass on the wisdom!
[2020-11-18T22:32:21.243Z] Recording test results

Screen Shot 2020-11-18 at 6 10 38 PM

**Comment by mdelapenya** [comment]
About the pre-submit errors, they are related to gherkin lint. Could you fix this line? https://github.com/elastic/e2e-testing/blob/049579d7538103f2f66350586b4f434bd9d12117/e2e/_suites/fleet/features/fleet_mode_agent.feature#L46

From:
```gherkin
  When the "elastic-agent" process is "restarted" on the host
    And the "filebeat" process is in the "started" state on the host
```
To:
```gherkin
  When the "elastic-agent" process is "restarted" on the host
  Then the "filebeat" process is in the "started" state on the host
```
>This is not related to your changes, so it will fix it for any other PR touching that file, which would fail in any case.

**Comment by mdelapenya** [comment]
To run the tests with a GH comment, you can check: https://github.com/elastic/e2e-testing/blob/master/.ci/Jenkinsfile#L27

**Comment by EricDavisX** [comment]
@mdelapenya its ready for review, but I had to feel-out how the linter would react.  Can you help point me to the linting guidelines I can read thru?

File: e2e/_suites/fleet/features/fleet_mode_agent.feature
Comment (mdelapenya) [inline e2e/_suites/fleet/features/fleet_mode_agent.feature:83]: I believe this is part of the `When` clause, as it's what triggers the `Then` parts
```suggestion
    And the "elastic-agent" process is "stopped" on the host
```

File: e2e/_suites/fleet/features/fleet_mode_agent.feature
Comment (mdelapenya) [inline e2e/_suites/fleet/features/fleet_mode_agent.feature:85]: The outcomes for the `When` start here
```suggestion
  Then the "elastic-agent" process is "started" on the host
```

**Review by mdelapenya — CHANGES_REQUESTED** [review]

File: e2e/_suites/fleet/features/fleet_mode_agent.feature
Comment (mdelapenya) [reply e2e/_suites/fleet/features/fleet_mode_agent.feature:83]: With current sequence, we are telling that if we un-enroll the agent, then the process is stopped. If that is correct, then your scenario is perfect.

File: e2e/_suites/fleet/features/fleet_mode_agent.feature
Comment (mdelapenya) [reply e2e/_suites/fleet/features/fleet_mode_agent.feature:85]: This is what we want to verify, that the agent is started again, right?

File: e2e/_suites/fleet/features/stand_alone_agent.feature
Comment (EricDavisX) [reply e2e/_suites/fleet/features/stand_alone_agent.feature:1]: ah! I forgot we had hard-coded refs to tags, we shouldn't plan to change them often, so hopefully not a problem moving forward

File: e2e/_suites/fleet/features/fleet_mode_agent.feature
Comment (EricDavisX) [reply e2e/_suites/fleet/features/fleet_mode_agent.feature]: I don't know if it passed linting as it was... I'll talk to Manu off-line and figure out any disconnect

File: e2e/_suites/fleet/features/fleet_mode_agent.feature
Comment (EricDavisX) [reply e2e/_suites/fleet/features/fleet_mode_agent.feature]: no, you're right - I believe I had only changed it as a guess to meet linting requirments, it is more accurate the way it was, I have revereted it.  And it passes linting, so it must have been changed during that period where I was unsure of the linting rules and I went overboard hoping to get it to pass.  it passes locally as such, if it fails in ci - then we have a mis-match in how the rules are being applied somehow, which we can follow up as needed.

to finish off that part of the convo, I broke linting to make sure I got some feedback... and I saw the local lint error, then I changed it to the syntax i've submitted with my last / latest comment and it is back to passing.

Screen Shot 2020-11-30 at 4 00 20 PM

**Comment by EricDavisX** [comment]
should be all good to go now, if not, we can review what failed (test or lint!)

**Review by mdelapenya — APPROVED** [review]
Hey @EricDavisX this LGTM! Great job with the Gherkin sequencing! Let's see how it goes with test failures and flakiness before merging this.

Let's also discuss (maybe in another issue) if we move the `systemd-specific` scenario to a separate feature file, not added to the CI parallel branches.

**Comment by EricDavisX** [comment]
Hi, I've pushed up latest changes which:
- add general support for 'skipping' tests marked as `@skip` in CI.
- added a skip tag to the new rpm / deb tests as they are concretely failing and we have a ticket logged
- that ticket is here: https://github.com/elastic/beats/issues/22744

here is evidence of my validating the skip works:
with the 'skip' as:
`SUITE="fleet" TAGS="test && ~skip" DEVELOPER_MODE=true TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE make -C e2e functional-test
`
passing-the-~skip-flag

without the skip:
NOT-passing-the~skip-flag

source file for test ref:
2-scenarios-one-is-skipped
 
Manu this also closes the 'skip' support request as logged in:
https://github.com/elastic/e2e-testing/issues/490

**Pull request merged**
